org.uqbar.project.wollok.lib/src/wollok/lang/WDictionary.xtend

```diff
@@ -0,0 +1,72 @@
+package wollok.lang
+
+import java.util.Map
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.uqbar.project.wollok.interpreter.api.WollokInterpreterAccess
+import org.uqbar.project.wollok.interpreter.core.WollokObject
+import org.uqbar.project.wollok.interpreter.nativeobj.JavaWrapper
+import org.uqbar.project.wollok.interpreter.nativeobj.NativeMessage
+import static extension org.uqbar.project.wollok.interpreter.nativeobj.WollokJavaConversions.*
+import static extension org.uqbar.project.wollok.lib.WollokSDKExtensions.*
+import java.util.HashMap
+
+/**
+ * Native part of the wollok.lang.List class
+ * 
+ * @author flbulgarelli
+ */
+class WDictionary implements JavaWrapper<Map<WollokObject, WollokObject>> {
+	@Accessors var Map<WollokObject, WollokObject> wrapped
+	protected extension WollokInterpreterAccess = new WollokInterpreterAccess
+
+	new() {
+		wrapped = newHashMap
+	}
+
+	def void clear() { wrapped.clear }
+
+	@NativeMessage("equals")
+	def wollokEquals(WollokObject other) {
+		other.hasNativeType(this.class.name) && (other.getNativeObject(this.class).wrapped == this.wrapped)
+	}
+
+	@NativeMessage("==")
+	def wollokEqualsEquals(WollokObject other) { wollokEquals(other) }
```

Comment:
This can be written in wollok itself. It just delegates.
I believe it's already implemented like that in wollok.lang.Object.
So only one should be overriden natively.

Actually I don't think we need to re implement the equals natively, if it is already implemented in Object based on identity